drivers/flash/Kconfig

```diff
@@ -45,6 +52,7 @@ config FLASH_PAGE_LAYOUT
 	help
 	  Enables API for retrieving the layout of flash memory pages.
 
+if !FLASH_HAS_SUBSYS_DRIVER
```

Comment:
I'd like to avoid wrapping all the in-tree Kconfig sources with this. But I suspect we have no choice ... I'll get back to you about our options.

Reply:
It's not the best, but this is the cleanest option I've found.

Reply:
Yeah, I can't find anything cleaner either.

This will have to do until we have better Kconfig tooling.